MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift

```diff
@@ -0,0 +1,156 @@
+extension MiniAppScriptMessageHandler {
+    func requestCustomPermissions(requestParam: RequestParameters?, callbackId: String) {
+        cachedUnknownCustomPermissionRequest.removeAll()
+        userRespondedRequestList.removeAll()
+        guard let requestParamValue = requestParam?.customPermissions else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppCustomPermissionError(customPermissionError: .invalidCustomPermissionRequest))
+            return
+        }
+
+        guard let miniAppPermissionRequestModelList = prepareCustomPermissionsRequestModelList(
+            permissionList: requestParamValue),
+            miniAppPermissionRequestModelList.count > 0 else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppCustomPermissionError(customPermissionError: .invalidCustomPermissionsList))
+            return
+        }
+        checkCustomPermissionsRequestStatusInCache(miniAppPermissionRequestModelList: miniAppPermissionRequestModelList, callbackId: callbackId)
+    }
+
+    func checkCustomPermissionsRequestStatusInCache(miniAppPermissionRequestModelList: [MASDKCustomPermissionModel], callbackId: String) {
+        /// storedPermissionsList - List of Permissions that is already stored in the cache
+        if let storedPermissionsList = self.miniAppStatus.getCustomPermissions(forMiniApp: self.miniAppId) {
+            /// filteredRequestModelList - List of Permissions that user hasn't responded yet.
+            let userNotRespondedRequestList = miniAppPermissionRequestModelList.filter {
+                !storedPermissionsList.contains($0)
+            }
+            userRespondedRequestList = storedPermissionsList.filter { miniAppPermissionRequestModelList.contains($0)
+            }
+            /// If the User hasn't responded to more than one permission that is requested, request Host app to request Custom Permissions
+            if userNotRespondedRequestList.count > 0 {
+                requestHostApp(customPermissionRequestList: userNotRespondedRequestList, callbackId: callbackId)
+            } else {
+                /// If the user has already responded to all the permissions that is requested filteredRequestModelList
```

Comment:
Same here, I think this comment isn't needed. It seems like the kind of comment that could quickly become outdated and not get updated, so it's better to let the code speak for itself.

Reply:
This was a Big PR and I felt like adding a few comments that would help the reviewers. I was planning to remove them before merging to the master